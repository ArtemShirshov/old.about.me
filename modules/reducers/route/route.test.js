import {getRoute, setPage, route, RouteFixture} from './index';

describe('route reducer', () => {
    it('should return the initial state', () => {
        // eslint-disable-next-line no-undefined
        expect(route(undefined, {})).toMatchSnapshot();
    });

    it('should handle "setPage" action', () => {
        const action = {type: setPage, payload: RouteFixture};
        expect(route({}, action)).toMatchSnapshot();
    });
});

describe('Action creators for "route" reducer', () => {
    test('action "getRoute"', () => {
        expect(getRoute('/test')).toMatchSnapshot();
    });

    test('action "setPage"', () => {
        expect(setPage(RouteFixture)).toMatchSnapshot();
    });
});
