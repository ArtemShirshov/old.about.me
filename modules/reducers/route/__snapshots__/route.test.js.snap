// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action creators for "route" reducer action "getRoute" 1`] = `
Object {
  "payload": Object {
    "request": Object {
      "baseURL": "https://api.gusadev.com/develop/",
      "method": "GET",
      "url": "route/14/test",
    },
  },
  "type": "GET_ROUTE",
}
`;

exports[`Action creators for "route" reducer action "setPage" 1`] = `
Object {
  "payload": Object {
    "page": "category",
    "params": Object {
      "id": 1000,
    },
  },
  "type": "SET_PAGE",
}
`;

exports[`route reducer should handle "setPage" action 1`] = `
Object {
  "page": "category",
  "params": Object {
    "id": 1000,
  },
}
`;

exports[`route reducer should return the initial state 1`] = `
Object {
  "page": "",
  "params": Object {},
}
`;
